kimchi/src/prover.rs

```diff
@@ -711,17 +747,31 @@ where
                         .iter()
                         .map(|c| c.eval(e, index.max_poly_size))
                         .collect(),
-                    table: lcs
-                        .lookup_table
-                        .iter()
-                        .map(|p| p.eval(e, index.max_poly_size))
-                        .rev()
-                        .fold(vec![Fr::<G>::zero()], |acc, x| {
-                            acc.into_iter()
-                                .zip(x.iter())
-                                .map(|(acc, x)| acc * joint_combiner + x)
-                                .collect()
-                        }),
+                    table: {
+                        let base_table = lcs
+                            .lookup_table
+                            .iter()
+                            .map(|p| p.eval(e, index.max_poly_size))
+                            .rev()
+                            .fold(vec![Fr::<G>::zero()], |acc, x| {
+                                acc.into_iter()
+                                    .zip(x.iter())
+                                    .map(|(acc, x)| acc * joint_combiner + x)
+                                    .collect()
+                            });
```

Comment:
btw I think it'd be nice to move all the similar (join columns with joint_combiner code) to a single place behind some abstraction. Could open an issue about that